react/components/CheckboxGroup/index.js

```diff
@@ -0,0 +1,117 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import some from 'lodash/some'
+import every from 'lodash/every'
+
+import Checkbox from '../Checkbox'
+
+class CheckboxGroup extends Component {
+  constructor(props) {
+    super(props)
+    this.state = { groupChecked: false }
+  }
+
+  isPartialChecked = () => {
+    const { checkedMap } = this.props
+    const checkedValues = Object.values(checkedMap).map(value => value.checked)
+    return some(checkedValues) && !every(checkedValues)
+  }
+
+  isAllChecked = () => {
+    const { checkedMap } = this.props
+    const checkedValues = Object.values(checkedMap).map(value => value.checked)
+    return every(checkedValues)
+  }
+
+  handleOnChange = key => {
+    const { checkedMap } = this.props
+    const newCheckedMap = {
+      ...checkedMap,
+      [key]: { ...checkedMap[key], checked: !checkedMap[key].checked },
+    }
+    this.props.onGroupChange(newCheckedMap)
+  }
+
+  handleOnGroupChange = () => {
+    if (this.isPartialChecked() || this.isAllChecked()) {
+      const { checkedMap } = this.props
+      const newCheckedMap = { ...checkedMap }
+      Object.keys(checkedMap).forEach(
+        key => (newCheckedMap[key].checked = false)
+      )
+      this.props.onGroupChange(newCheckedMap)
+    } else {
+      const { checkedMap } = this.props
+      const newCheckedMap = { ...checkedMap }
+      Object.keys(checkedMap).forEach(
+        key => (newCheckedMap[key].checked = true)
```

Comment:
:pick: 

```suggestion
        key => { newCheckedMap[key].checked = true }
```

Reply:
⛏ This block is being duplicated as well. Perhaps it could be a function like `setGroupChecked(value: boolean)`?